packages/p2p/src/components/modal-manager/modal-manager-context-provider.jsx

```diff
@@ -23,15 +23,29 @@ const ModalManagerContextProvider = props => {
      * @param {Object|Object[]} modals - list of object modals to set props, each modal object must contain a 'key' attribute and 'props' attribute
      */
     const useRegisterModalProps = modals => {
+        const registered_modals = [];
+
         const registerModals = React.useCallback(() => {
             if (Array.isArray(modals)) {
-                modals.forEach(modal => setModalProps(modal_props.set(modal.key, modal.props)));
+                modals.forEach(modal => {
```

Comment:
use `registered_modals` to keep track of which modals has registered their props, and only de-register these props if the component is unmounted